darts/tests/optional_deps/test_onnx.py

```diff
@@ -0,0 +1,213 @@
+from itertools import product
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+import pytest
+
+import darts.utils.timeseries_generation as tg
+from darts import TimeSeries
+from darts.tests.conftest import ONNX_AVAILABLE, TORCH_AVAILABLE, tfm_kwargs
+from darts.utils.onnx_utils import prepare_onnx_inputs
+
+if not (TORCH_AVAILABLE and ONNX_AVAILABLE):
+    pytest.skip(
+        f"Torch or Onnx not available. {__name__} tests will be skipped.",
+        allow_module_level=True,
+    )
+import onnx
+import onnxruntime as ort
+
+from darts.models import (
+    BlockRNNModel,
+    NBEATSModel,
+    NHiTSModel,
+    NLinearModel,
+    TiDEModel,
+    TransformerModel,
+)
+
+# TODO: check how RINorm can be handled with respect to ONNX
+torch_model_cls = [
+    BlockRNNModel,
+    NBEATSModel,
+    NHiTSModel,
+    NLinearModel,
+    # TCNModel,
+    # TFTModel,
+    TiDEModel,
+    TransformerModel,
+    # TSMixerModel,
```

Comment:
Very nice tests :)

let's use only BlockRNN, NHiTS and TiDE to reduce from 40 seconds to 5.

Also use the fast dev run again:

```
tfm_kwargs = copy.deepcopy(tfm_kwargs)
tfm_kwargs["pl_trainer_kwargs"]["fast_dev_run"] = True
```